sdk/src/main/java/io/radar/sdk/RadarBatteryManager.kt

```diff
@@ -0,0 +1,97 @@
+package io.radar.sdk
+
+import android.annotation.SuppressLint
+import android.app.usage.UsageStatsManager
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.os.BatteryManager
+import android.os.Build
+import android.os.PowerManager
```

Comment:
do you have any sense for what the runtimes are on these native APIs? we can see how it works in the wild, but my concern is that they'll add more latency to scheduling / handling jobs

Reply:
@jsani-radar if we wait and merge #142, there's less concern - since that moves all the logging calls to a separate thread. I do not have a good sense of what additional time this would add, however my assumption is it would be pretty small. Android is very good about preventing hangs and ANRs.

Reply:
This is related to @tjulien's [comment](https://github.com/radarlabs/radar-sdk-android/pull/154#discussion_r779592372) related to keeping these extra APIs behind test keys so we can dogfood it